src/sagemaker/predictor.py

```diff
@@ -132,15 +132,18 @@ def delete_model(self):
         """Deletes the Amazon SageMaker models backing this predictor.
 
         """
-        for model_name in self._model_names:
-            self.sagemaker_session.delete_model(model_name)
+        try:
+            for model_name in self._model_names:
+                self.sagemaker_session.delete_model(model_name)
+        except Exception:
+            raise Exception('One or more models cannot be deleted, the deletion is incomplete.')
```

Comment:
say if we have 3 models and the second one failed do we want to try to delete the third one?